sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -155,6 +155,31 @@ def test_remove_zero_eig():
     assert_equal(Xt.shape, (3, 0))
 
 
+def test_leave_zero_eig():
+    """This test checks that fit().transform() returns the same result than
+    fit_transform() in case of non-removed zero eigenvalue"""
```

Comment:
Please add a comment indicating that this is a non-regression test, linking to this PR (or the corresponding issue)

Reply:
ok